relay-server/src/utils/rate_limits.rs

```diff
@@ -152,6 +155,10 @@ impl EnvelopeSummary {
                 summary.has_plain_attachments = true;
             }
 
+            if item.metrics_extracted() {
+                summary.metrics_extracted = true;
```

Comment:
Both sessions and transactions use this item header. Even if they are never sent in the same envelope, I would be defensive here and check the item type.